modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/LogsIndexModeCustomSettingsIT.java

```diff
@@ -0,0 +1,296 @@
+                    }
+                  }
+                }
+              }
+            }""";
+
+        assertOK(putComponentTemplate(client, "logs@custom", ignoreMalformedOverrideTemplate));
+        assertOK(createDataStream(client, "logs-custom-dev"));
+
+        var ignoreMalformedIndexSetting = (String) getSetting(
+            client,
+            getDataStreamBackingIndex(client, "logs-custom-dev", 0),
+            "index.mapping.ignore_malformed"
+        );
+        assertThat(ignoreMalformedIndexSetting, equalTo("false"));
+    }
+
+    public void testOverrideIgnoreDynamicBeyondLimit() throws IOException {
+        var ignoreMalformedOverrideTemplate = """
+            {
+              "template": {
+                "settings": {
+                  "index": {
+                    "mapping": {
+                      "total_fields": {
+                        "ignore_dynamic_beyond_limit": false
+                      }
+                    }
+                  }
+                }
+              }
+            }""";
+
+        assertOK(putComponentTemplate(client, "logs@custom", ignoreMalformedOverrideTemplate));
+        assertOK(createDataStream(client, "logs-custom-dev"));
+
+        var ignoreDynamicBeyondLimitIndexSetting = (String) getSetting(
+            client,
+            getDataStreamBackingIndex(client, "logs-custom-dev", 0),
+            "index.mapping.total_fields.ignore_dynamic_beyond_limit"
+        );
+        assertThat(ignoreDynamicBeyondLimitIndexSetting, equalTo("false"));
+    }
+
+    public void testAddNonCompatibleMapping() throws IOException {
+        var nonCompatibleMappingAdditionTemplate = """
+            {
+              "template": {
+                "mappings": {
+                  "properties": {
+                    "bomb": {
+                      "type": "ip",
+                      "doc_values": false
+                    }
+                  }
+                }
+              }
+            }""";
+
+        Exception e = assertThrows(ResponseException.class, () -> putComponentTemplate(client, "logs@custom", nonCompatibleMappingAdditionTemplate));
+        assertThat(
+            e.getMessage(),
+            containsString("updating component template [logs@custom] results in invalid composable template [logs]")
+        );
+        assertThat(e.getMessage(), containsString("field [bomb] of type [ip] doesn't support synthetic source because it doesn't have doc values"));
+    }
+
+    private static Map<String, Object> getMapping(final RestClient client, final String indexName) throws IOException {
+        final Request request = new Request("GET", "/" + indexName + "/_mapping");
+
+        Map<String, Object> mappings = ((Map<String, Map<String, Object>>) entityAsMap(client.performRequest(request)).get(indexName)).get(
+            "mappings"
+        );
+
+        return mappings;
+    }
+
+    private Function<Object, Map<String, Object>> subObject(String key) {
```

Comment:
This is a pretty cool trick for working with semi-structured stuff in `Map<String, Object>`.

Reply:
I'll merge this as is and we can move it.